File: src/main/java/org/example/expert/domain/todo/repository/TodoQueryRepository.java
package org.example.expert.domain.todo.repository;

import java.time.LocalDate;
import java.util.Optional;

import org.example.expert.domain.todo.dto.response.TodoSummaryResponse;
import org.example.expert.domain.todo.entity.Todo;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TodoQueryRepository {

	Optional<Todo> findByIdWithUser(Long todoId);

	Page<TodoSummaryResponse> searchTodosWithFilters(String keyword, String nickname, LocalDate startDate, LocalDate endDate, Pageable pageable);
}

File: src/main/kotlin/com/example/domain/todo/repository/TodoQueryRepository.kt
package com.example.domain.todo.repository

import com.example.domain.todo.dto.response.TodoSummaryResponse
import com.example.domain.todo.entity.Todo
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.LocalDate
import java.util.*

interface TodoQueryRepository {
    fun findByIdWithUser(todoId: Long): Optional<Todo>

    fun searchTodosWithFilters(
        keyword: String?,
        nickname: String?,
        startDate: LocalDate?,
        endDate: LocalDate?,
        pageable: Pageable
    ): Page<TodoSummaryResponse>
}
